orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/messages.kt

```diff
@@ -73,10 +73,22 @@ data class CompleteTask(
   override val application: String,
   override val stageId: String,
   override val taskId: String,
-  val status: ExecutionStatus
+  val status: ExecutionStatus,
+  val originalStatus: ExecutionStatus = status
```

Comment:
Tracking the original / unmodified task status

I don't particularly like this _but_ I'm not sure it's avoidable given the calculation happening in `RunTaskHandler`. It feels like we could migrate to a point where the logic to override a task status is in the `CompleteTaskHandler` instead.

Can you think of a more elegant way to do this @robfletcher?

Reply:
I like the idea of having `CompleteTaskHandler` decide. Could we then just use the `isComplete` on `ExecutionStatus` to decide whether to run the next task?